api/src/services/auth.service.ts

```diff
@@ -84,11 +85,74 @@ export class AuthService {
     res: Response,
     user: User,
     incomingRefreshToken?: string,
+    reCaptchaToken?: string,
+    reCaptchaConfigured?: boolean,
+    mfaCode?: boolean,
+    shouldReCaptchaBlockLogin?: boolean,
   ): Promise<SuccessDTO> {
     if (!user?.id) {
       throw new UnauthorizedException('no user found');
     }
 
+    if (reCaptchaConfigured && !user.mfaEnabled && !mfaCode) {
+      const client = new RecaptchaEnterpriseServiceClient({
+        credentials: {
+          private_key: process.env.GOOGLE_API_KEY.replace(/\\n/gm, '\n'),
+          client_email: process.env.GOOGLE_API_EMAIL,
+        },
+        projectID: process.env.GOOGLE_API_ID,
+      });
+
+      const request = {
+        assessment: {
+          event: {
+            token: reCaptchaToken,
+            siteKey: process.env.RECAPTCHA_KEY,
+          },
+        },
+        parent: client.projectPath(process.env.GOOGLE_CLOUD_PROJECT_ID),
+      };
+
+      const [response] = await client.createAssessment(request);
+      client.close();
+
+      if (!response.tokenProperties.valid && shouldReCaptchaBlockLogin) {
+        throw new UnauthorizedException({
+          name: 'failedReCaptchaToken',
+          knownError: true,
+          message: `The ReCaptcha CreateAssessment call failed because the token was: ${response.tokenProperties.invalidReason}`,
+        });
+      }
+
+      if (response.tokenProperties.action === 'login') {
+        response.riskAnalysis.reasons.forEach((reason) => {
+          console.log(reason);
+        });
+
+        console.log(`The ReCaptcha score is ${response.riskAnalysis.score}`);
```

Comment:
I know this is here for testing purposes, but long term we probably wouldn't want this cluttering the logs

Reply:
Yep yep, after we collect an average staging/prod score we can remove this!